### PR TITLE
[DO NOT MERGE] Add xdg-toplevel-decoration support

### PR DESCRIPTION
**Pull request by emersion**
This adds support for xdg-toplevel-decoration in rootston. The preferred mode is "client". rootston will accept the client's mode requests.

A new example client is added. When launched, it will be yellow if using server-side decorations and green if using client-side decorations. The client's initial preferred mode can be given as an argument (either "client" or "server"). The client's preferred mode can be toggled by clicking on its surface.

Test plan: try the example

### CODE REVIEW

**Comment by BurningTreeC** [comment]
I like copy-pasta!

**Comment by emersion** [comment]
Just look at the last 3 commits f you don't want to review the xdg-shell changes.

**Comment by emersion** [comment]
This PR is ready for review.

File: examples/toplevel-decoration.c [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	wl_display_roundtrip(display);
+
+	draw();
+
+	while (wl_display_dispatch(display) != -1) {
+		// No-op
+	}
+
+	return EXIT_SUCCESS;
+}
```
Comment (SirCmpwn) [inline examples/toplevel-decoration.c]: Would be nice if the example could be made to toggle CSD at runtime

Comment (emersion) [reply examples/toplevel-decoration.c]: Just click on the window, it will switch the mode.

**Comment by Sunderland93** [comment]
This is a replace of KDE's Server-Decoration protocol, right?

**Comment by SirCmpwn** [comment]
It is, yes, developed in cooperation with KDE and Mir.

**Comment by Sunderland93** [comment]
@SirCmpwn will it be added to wayland-protocols?

**Comment by SirCmpwn** [comment]
We have sent it in for inclusion, yeah.

**Comment by emersion** [comment]
Superseded by #1053

**Pull request closed without merging**